lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -695,6 +695,12 @@ class ExprContextAnalyzer {
       break;
     }
     default:
+      if (auto *AFD = dyn_cast<AbstractFunctionDecl>(D)) {
+        assert(isSingleExpressionBodyForCodeCompletion(AFD->getBody()));
+        singleExpressionBody = true;
+        recordPossibleType(getReturnTypeFromContext(AFD));
```

Comment:
Please ensure this doesn't work for `init`/`deinit`. i.e.
```swift
enum MyEnum {
  case foo
  init() { .#^HERE^# }
}
```
should *not* work.

Reply:
Ah good point, for "init?" it should technically show `.none` and `nil`.